File: aoc/2021/day-2/1.py
with open('aoc/day-2/input.txt') as f:
  ans = 0
  data = [i.split() for i in f.read().splitlines()]
  horz = 0
  depth = 0
  for i, x in data:
    print(i, x)
    if i == "forward":
      horz += int(x)
    elif i == "up":
      depth -= int(x)
    elif i == "down":
      depth += int(x)
  print(horz * depth)

File: aoc/2021/day-2/1.exs
defmodule Solution do
  def part1() do
    File.read!("input.txt")
    |> String.split("\n", trim: true)
    |> Enum.reduce({0, 0}, fn
      "forward " <> number, {depth, position} ->
        {depth, String.to_integer(number) + position}

      "down " <> number, {depth, position} ->
        {depth + String.to_integer(number), position}

      "up " <> number, {depth, position} ->
        {depth - String.to_integer(number), position}
    end)
    |> then(fn {depth, position} -> depth * position end)
  end

  def part2() do
    File.read!("input.txt")
    |> String.split("\n", trim: true)
  end
end

defmodule Submarine do
  def navigate("forward " <> number, {depth, position, aim}) do
    number = String.to_integer(number)
    {depth + aim * number, position + number, aim}
  end
end

IO.puts(Solution.part1())
